src/hb-ot-glyf-table.hh

```diff
@@ -81,26 +80,170 @@ struct glyf
+    char *loca_prime_data = (char *) calloc(1, loca_prime_size);
+
+    if (unlikely (!loca_prime_data)) return false;
+
+    if (use_short_loca)
+      _write_loca <decltype (padded_offsets), HBUINT16> (padded_offsets, 1, hb_array_t<HBUINT16> ((HBUINT16*) loca_prime_data, padded_offsets.len ()));
+    else
+      _write_loca <decltype (padded_offsets), HBUINT32> (padded_offsets, 0, hb_array_t<HBUINT32> ((HBUINT32*) loca_prime_data, padded_offsets.len ()));
+
+    hb_blob_t * loca_blob = hb_blob_create (loca_prime_data,
+					    loca_prime_size,
+					    HB_MEMORY_MODE_READONLY,
+					    loca_prime_data,
+					    free);
+
+    plan->add_table (HB_OT_TAG_loca, loca_blob);
+    _add_head_and_set_loca_version(plan, use_short_loca);
+
+    hb_blob_destroy (loca_blob);
+    return true;
+  }
+
+  template<typename IteratorIn, typename EntryType, typename IteratorOut,
+	   hb_requires (hb_is_source_of (IteratorIn, unsigned int)),
+	   hb_requires (hb_is_source_of (IteratorOut, EntryType))>
+  static void
+  _write_loca (IteratorIn it, unsigned right_shift, IteratorOut dest)
+  {
+    unsigned int offset = 0;
+    + it
+    | hb_map ([&] (unsigned int padded_size) {
+      unsigned int result = offset >> right_shift;
+      DEBUG_MSG(SUBSET, nullptr, "loca entry offset %d shifted %d", offset, result);
+      offset += padded_size;
+      return result;
+    })
+    | hb_sink (dest);
+  }
+
+  // requires source of SubsetGlyph complains the identifier isn't declared
+  template <typename Iterator>
+  bool serialize(hb_serialize_context_t *c,
+		 Iterator it,
+		 const hb_subset_plan_t *plan)
   {
-    hb_blob_t *glyf_prime = nullptr;
-    hb_blob_t *loca_prime = nullptr;
-
-    bool success = true;
-    bool use_short_loca = false;
-    if (hb_subset_glyf_and_loca (plan, &use_short_loca, &glyf_prime, &loca_prime)) {
-      success = success && plan->add_table (HB_OT_TAG_glyf, glyf_prime);
-      success = success && plan->add_table (HB_OT_TAG_loca, loca_prime);
-      success = success && _add_head_and_set_loca_version (plan, use_short_loca);
-    } else {
-      success = false;
+    TRACE_SERIALIZE (this);
+
+    + it
+    | hb_apply ( [&] (const SubsetGlyph& _) { _.serialize (c, plan); });
+
+    return_trace (true);
+  }
+
+  bool subset (hb_subset_context_t *c) const
+  {
+    TRACE_SUBSET (this);
+
+    glyf *glyf_prime = c->serializer->start_embed <glyf> ();
+    if (unlikely (!glyf_prime)) return_trace (false);
+
+    // Byte region(s) per glyph to output
+    // unpadded, hints removed if so requested
+    // If we fail to process a glyph we produce an empty (0-length) glyph
+    hb_vector_t<SubsetGlyph> glyphs;
+    _populate_subset_glyphs (c->plan, &glyphs);
+
+    glyf_prime->serialize (c->serializer, hb_iter (glyphs), c->plan);
+
+    hb_vector_t<unsigned int> padded_offsets;
+    + hb_iter (glyphs)
+    | hb_map ([&] (const SubsetGlyph& _) { return _.padded_size(); })
+    | hb_sink (padded_offsets);
```

Comment:
Semi-colon next line.  Maybe I give up on  that. :)

Reply:
It looks cooler, I like it lolz